tests/unit/field/test_fieldperp.cxx

```diff
@@ -825,6 +825,18 @@ TEST_F(FieldPerpTest, AddFieldPerpField2D) {
   EXPECT_TRUE(IsFieldEqual(c, 3.0));
 }
 
+TEST_F(FieldPerpTest, AddField2DFieldPerp) {
```

Comment:
Thanks for this, and all the new tests! It would be good to test the cases with `BoutReal` as one of the arguments, as there is some special handling in these cases (e.g. getting `localmesh`).

Reply:
The tests were pretty complete already; the tests for `BoutReal` operators are already there :-) The only ones I had to add were for operators that were previously not implemented.

Reply:
Ah great! Thanks @johnomotani